libc/string/string_tokpbrkstr.c

```diff
@@ -0,0 +1,1019 @@
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("cdefg", "abc"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abefg", "cde"));
+}
+
+
+TEST(string_str, big_string)
+{
+	int i;
+	char str[PATH_MAX];
+
+	memset(str, 1, sizeof(str));
+
+	for (i = 1; i < PATH_MAX - 1; i++) {
+		str[i] = 2;
+		TEST_ASSERT_EQUAL_PTR(&str[i], strstr(str, "\2"));
+		TEST_ASSERT_EQUAL_PTR(str, strstr(str, "\1"));
+
+		str[i] = 1;
+	}
+}
+
+
+/*
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(string_pbrk)
+{
+}
+
+
+TEST_TEAR_DOWN(string_pbrk)
+{
+}
+
+
+TEST(string_pbrk, basic)
+{
+	const char loremIpsum[BUFFSIZE] = LOREM_IPSUM;
+	char *asciiStr;
+
+	asciiStr = testdata_createCharStr(ASCII_LENGTH);
+
+	TEST_ASSERT_EQUAL_PTR(loremIpsum, strpbrk(loremIpsum, "Lorem"));
+
+	/*
+	 * strpbrk is sensitive to all elements in the array where
+	 * is looking up which means 'm' is the first element of array 'loremIpsum' that strpbrk found
+	 */
+	TEST_ASSERT_EQUAL_PTR(&loremIpsum[4], strpbrk(loremIpsum, "Ipsum"));
+
+	/* Like in the case before strpbrk found 'o' at second place in the array and his output point at second place in the array*/
+	TEST_ASSERT_EQUAL_PTR(&loremIpsum[1], strpbrk(loremIpsum, "Dolor"));
+
+	/* Simple usage of strpbrk with ascii set */
+	TEST_ASSERT_EQUAL_PTR(asciiStr, strpbrk(asciiStr, asciiStr));
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[ASCII_LENGTH / 4], strpbrk(asciiStr, &asciiStr[ASCII_LENGTH / 4]));
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[ASCII_LENGTH / 3], strpbrk(asciiStr, &asciiStr[ASCII_LENGTH / 3]));
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[ASCII_LENGTH / 2], strpbrk(asciiStr, &asciiStr[ASCII_LENGTH / 2]));
+
+	free(asciiStr);
+}
+
+
+TEST(string_pbrk, empty_args)
+{
+	/* Different scenarios to acquire null pointer as output*/
+	TEST_ASSERT_NULL(strpbrk("", "abc"));
+	TEST_ASSERT_NULL(strpbrk("", ""));
+	TEST_ASSERT_NULL(strpbrk("abc", ""));
+}
+
+
+TEST(string_pbrk, out_of_ascii)
+{
+	int i;
+
+	char *extAsciiStr = create_extAscii_set();
+
+	/* Checking the ability of strpbrk to use chars from 0 to 255 as elements of arrays in search*/
+	for (i = 0; i < sizeof(extAsciiStr) - 1; i++) {
+		TEST_ASSERT_EQUAL_PTR(&extAsciiStr[i], strpbrk(extAsciiStr, &extAsciiStr[i]));
+	}
+
+	free(extAsciiStr);
+}
+
+
+TEST(string_pbrk, strpbrk_order)
+{
+	int i;
+	char reversStr[ASCII_LENGTH],
+		*testStr,
+		str[] = "abc";
+
+	testStr = testdata_createCharStr(ASCII_LENGTH);
+
+	TEST_ASSERT_NOT_NULL(reversStr);
+
+	for (i = 0; i < ASCII_LENGTH - 1; i++) {
+		reversStr[i] = testStr[ASCII_LENGTH - i - 2];
+	}
+	reversStr[ASCII_LENGTH - 1] = 0;
+
+	TEST_ASSERT_EQUAL_STRING(str, strpbrk("abc", "cba"));
+	TEST_ASSERT_EQUAL_STRING(str, strpbrk("abc", "bac"));
+	TEST_ASSERT_EQUAL_STRING(str, strpbrk("abc", "bca"));
+
+	/* Asserting reversed string */
+	for (i = 0; i < ASCII_LENGTH - 1; i++) {
+		TEST_ASSERT_EQUAL_PTR(testStr, strpbrk(testStr, &reversStr[i]));
+	}
+
+	free(testStr);
+}
+
+
+TEST(string_pbrk, part_of_str)
```

Comment:
I think we can split this case to `multiple` one, where you could check passing same bytes multiple time and to `additional_bytes` that could check passing some bytes not present in s1 to s2 (you did it here). Moreover, you can add a third case called `not_present` or sth like that, where any of bytes in s2 is not present in s1.